src/services/services.ts

```diff
@@ -3345,6 +3347,20 @@ export function createLanguageService(
         );
     }
 
+    function getImports(fileName: string): readonly string[] {
+        synchronizeHostData();
+        const file = getValidSourceFile(fileName);
+        const imports: Set<string> = new Set();
+        let specifiers = file.imports;
+        specifiers = specifiers.slice(specifiers.findIndex(nodeIsSynthesized) + 1);
+        if (specifiers.length === 0) return [];
+        for (const specifier of specifiers) {
+            const name = program.getResolvedModuleFromModuleSpecifier(specifier, file)?.resolvedModule?.resolvedFileName;
+            if (name && !imports.has(name)) imports.add(name);
+        }
+        return Array.from(imports);

```

Comment:
use `arrayFrom` helper instead ?

Reply:
Why? It looks worse than the built-in, since it's an overloaded function that checks at each iteration whether a mapping function is defined.